### PR TITLE
Fix 1825

### PR DESCRIPTION
**Pull request by chaosphere2112**
Fixes the issue discovered by Ram on the mailing list, where you could not set the missing color for a graphics method using a color's name. Adds a test to cover that usecase.

### CODE REVIEW

**Comment by chaosphere2112** [comment]
@doutriaux1 @aashish24

File: Packages/vcs/Lib/utils.py [hunk 1]
```diff
@@ -1716,6 +1716,37 @@ def getworldcoordinates(gm, X, Y):
     return wc
 
 
+def rgba_color(color, colormap):
+    """Try all of the various syntaxes of colors and return 0-100 RGBA values."""
+    try:
+        # Is it a colormap index?
+        return colormap.index[color]
+    except ValueError:
```
Comment (aashish24) [inline Packages/vcs/Lib/utils.py]: should be  a IndexError?

File: Packages/vcs/Lib/utils.py [hunk 2]
```diff
@@ -1716,6 +1716,37 @@ def getworldcoordinates(gm, X, Y):
     return wc
 
 
+def rgba_color(color, colormap):
```
Comment (aashish24) [inline Packages/vcs/Lib/utils.py]: Could you please add some comments on this code behavior?

File: Packages/vcs/Lib/utils.py [hunk 3]
```diff
@@ -1716,6 +1716,37 @@ def getworldcoordinates(gm, X, Y):
...
+        return colormap.index[color]
+    except ValueError:
+        # Is it a color tuple?
+        if len(color) == 3 or len(color) == 4:
+            for c in color:
+                try:
+                    int(c)
+                except:
+                    break
+            else:
```
Comment (aashish24) [inline Packages/vcs/Lib/utils.py]: is it just me or this else is not aligned?

Comment (chaosphere2112) [inline Packages/vcs/Lib/utils.py]: No, the `else` is for the `for` loop; it executes if the code doesn't `break`.

File: Packages/vcs/Lib/utils.py (hunk 1 above)
Comment (chaosphere2112) [inline Packages/vcs/Lib/utils.py]: Nope, it's a `ValueError`. We implement `__setitem__` and `__getitem__` on the colormap object, and raise a `ValueError` if the key isn't a number between 0 and 256.

File: Packages/vcs/Lib/utils.py (hunk 3 above)
Comment (aashish24) [inline Packages/vcs/Lib/utils.py]: I see. Using try and except like this is consider best practice?

File: Packages/vcs/Lib/utils.py (hunk 1 above)
Comment (aashish24) [inline Packages/vcs/Lib/utils.py]: Interesting, I would expect a IndexError from this code but your code looks go to me.

**Comment by aashish24** [comment]
LGTM except that I have question on best practices for try and except. 

+2

File: Packages/vcs/Lib/utils.py (hunk 3 above)
Comment (chaosphere2112) [inline Packages/vcs/Lib/utils.py]: Yup, exceptions are cheap in python. [Ask for forgiveness, not permission](http://stackoverflow.com/a/12265860/725871)

**Pull request merged**